syntax = "proto3";

package paitoo;

service Analytics{
	
	rpc GetUploadedReviews(Query) returns (stream Review);

	rpc GetLikedReviews(Query) returns (stream Liked_Review);

	rpc GetFollowedRestaurants(Query) returns (stream Follow_Restaurant);

	rpc GetOpenedDish(Query) returns (stream Opened_Dish);
}

message Query{
	string queryMessage = 1;
}

// The record of a review
message Review {
	string uploader_id = 1;
	string review_id = 2;
	string restaurant_id = 3;
	int32 service = 4;
	int32 ambience = 5;
	int32 cleanliness = 6;
	string dish_name = 7;
	bool value_for_money = 8;
	int32 taste = 9;
}

// The record of when a user likes a review
message Liked_Review { 
	string user_id = 1;
	Review review = 2;
}
// The record of when a user follows a restaurant
message Follow_Restaurant {
	string user_id = 1;
	string restaurant_id = 2;
}

message Opened_Dish { 
	string user_id = 1;
	string restaurant_id = 2;
	string dishname = 3;
	string category = 4;
	bool searched = 5;
}